langchain/src/agents/initialize.ts

```diff
@@ -46,3 +47,68 @@ export const initializeAgentExecutor = async (
       throw new Error("Unknown agent type");
   }
 };
+
+export const initializeAgentExecutorWithOptions = async (
+  tools: Tool[],
+  llm: BaseLanguageModel,
+  options: {
+    agentType?: AgentType;
+    prefix?: string;
+    suffix?: string;
+    memory?: BaseMemory;
+    returnIntermediateSteps?: boolean;
+    verbose?: boolean;
+    callbackManager?: CallbackManager;
+  }
+): Promise<AgentExecutor> => {
+  const agentType = options.agentType ?? "zero-shot-react-description";
+  const returnIntermediateSteps = options.returnIntermediateSteps ?? false;
+  const verbose = options.verbose ?? false;
+  const callbackManager = options.callbackManager ?? getCallbackManager();
+
+  switch (agentType) {
+    case "zero-shot-react-description": {
+      return AgentExecutor.fromAgentAndTools({
+        agent: ZeroShotAgent.fromLLMAndTools(llm, tools, {
+          prefix: options.prefix,
+          suffix: options.suffix,
+        }),
+        tools,
+        returnIntermediateSteps,
+        verbose,
+        callbackManager,
+        memory: options.memory,
+      });
+    }
+    case "chat-zero-shot-react-description": {
+      return AgentExecutor.fromAgentAndTools({
+        agent: ChatAgent.fromLLMAndTools(llm, tools, {
+          prefix: options.prefix,
+          suffix: options.suffix,
+        }),
+        tools,
+        returnIntermediateSteps,
+        verbose,
+        callbackManager,
+        memory: options.memory,
+      });
+    }
+    case "chat-conversational-react-description": {
+      const executor = AgentExecutor.fromAgentAndTools({
+        agent: ChatConversationalAgent.fromLLMAndTools(llm, tools, {
+          prefix: options.prefix,
+          suffix: options.suffix,
+        }),
+        tools,
+        returnIntermediateSteps,
+        verbose,
+        callbackManager,
+        memory: options.memory,
```

Comment:
@nfcampos I removed the memory default here, but don't love the fact that we need to initialize memory with specific options like `memoryKey: "chat_history"`. Would it make sense to make a wrapper class with those defaults? 

Or even just change the memory key in `BufferMemory` to `chat_history`?

Reply:
This agent is "special" in that it actually is designed to work with memory, the other ones aren't. I agree the `chat_history` hidden requirement is bad, but we do need to default here to creating a memory

Reply:
I also changed the options input syntax for `initializeAgentExecutorWithOptions` to take a `promptArgs` option that will just get passed through directly to `fromLLMAndTools`, so we don't need to worry about conflating `systemMessage` and `prefix` - the user can look at the specific agent they're using and figure out which one is appropriate.

Longer term, subclassing at least the prompt args could make sense to me so we could do something like:
```js
promptArgs?: AgentCreatePromptArgs | ChatAgentCreatePromptArgs;
```
`ChatAgentCreatePromptArgs` could also take a memory key too and decouple the hidden reliance on a `memoryKey` set to `chat_history`.